catalog/cataloger.go

```diff
@@ -178,6 +183,28 @@ type CacheConfig struct {
 	Jitter  time.Duration
 }
 
+// CatalogerHooks describes the hooks available for some operations on the catalog.  Hooks are
+// called in a current transaction context; if they return an error the transaction is rolled
+// back.  Because these transactions are current, the hook can see the effect the operation only
+// on the passed transaction.
+type CatalogerHooks struct {
```

Comment:
Maybe add a branch filter by name/regex?
o.w. you're relying on the hook itself to filter by it.